dpytools/config/config.py

```diff
@@ -1,38 +1,81 @@
+from .properties.intproperty import IntegerProperty
+from .properties.string import StringProperty
 
 class Config:
+    
+    def __init__(self):
+        self._properties_to_validate: List[BaseProperty] = []
 
     @staticmethod
-    def from_env(config_dict: Dict[str, BaseProperty]):
-        # TODO = read in and populate property classes as
-        # per the example in the main readme.
-        # You need to populate with dot notation in mind so:
-        #
-        # StringProperty("fieldname", "fieldvalue")
-        #
-        # should be accessed on Config/self, so:
-        #
-        # value = config.fieldvalue.value
-        # i.e
-        # config.fieldvalue = StringProperty("fieldname", "fieldvalue")
-        #
-        # Worth looking at the __setattr_ dunder method and a loop
-        # for how to do this.
-        #
-        # Do track the BaseProperty's that you add ready for
-        # assert_valid_config call.
-        ...
+    def from_env(config_dict: Dict[str, Dict[str, Any]]) -> Config:
+
+        config = Config()
+
+        for env_var_name, value in config_dict.items():
+
+            value_for_property = os.environ.get(env_var_name, None)
+            assert value_for_property is not None, f'Required envionrment value "{env_var_name}" could not be found.'
+
+            if value["class"] == StringProperty:
+                if value["kwargs"]:
+                    regex = value["kwargs"].get("regex")
+                    min_len = value["kwargs"].get("min_len")
+                    max_len = value["kwargs"].get("max_len")
+                else:
+                    regex = None
+                    min_len = None
+                    max_len = None
+  
+                stringprop = StringProperty(
+                    _name = value["property"],
+                    _value = value_for_property,
+                    regex = regex,
+                    min_len = min_len,
+                    max_len = max_len
+                )
+
+                prop_name = value["property"]
+                setattr(config, prop_name, stringprop)
+                config._properties_to_validate.append(stringprop)
+
+            elif value["class"] == IntegerProperty:
+                if value["kwargs"]:
+                    min_val = value["kwargs"].get("min_val")
+                    max_val = value["kwargs"].get("max_val")
+                else:
+                    min_val = None
+                    max_val = None
+
+                intprop = IntegerProperty(
+                    _name = value["property"],
+                    _value = value_for_property,
+                    min_val = min_val,
+                    max_val = max_val
+                )
+
+                prop_name = value["property"]
+                setattr(config, prop_name, intprop)
+                config._properties_to_validate.append(intprop)
+
+            else:
+                raise TypeError("Incorrect value type specified for property assignment.")
```

Comment:
Good thinking, but can you add the type they did provide here for context, so something like:

```
Unsupported property type specified via "property" field, got <what they actual provided> should be on of StringProperty, IntegerProperty.
```

something like that, make it _really_ obvious what they've done wrong.